src/charm_state.py

```diff
@@ -4,18 +4,67 @@
 """This module defines the CharmState class which represents the state of the Flask charm."""
 
 import datetime
+import itertools
 import pathlib
 
 from ops.charm import CharmBase
+from pydantic import (  # pylint: disable=no-name-in-module
+    BaseModel,
+    Extra,
+    Field,
+    ValidationError,
+    validator,
+)
 
 from charm_types import WebserverConfig
+from exceptions import CharmConfigInvalidError
+
+
+class FlaskConfig(BaseModel, extra=Extra.allow):  # pylint: disable=too-few-public-methods
```

Comment:
The problem is that after validation, there's no place in the code that access each individual fields inside `flask_config`. That's why I choose to use dict instead of data class to reduce the unnecessary complexity of the interface.

Reply:
Ok let's keep it like this and re-evaluate if we make some changes